src/main/java/com/micetr0/mock_DB/MySqlDB.java

```diff
@@ -307,12 +376,68 @@ public void deleteAccount(String username) {
     }
 
     @Override
-    public void insertAccount(Account account) {
+    public Integer insertAccount(String username, String password) {
+        return executeTransaction((Connection conn) -> {
+            PreparedStatement instAccStmt = null;
+            PreparedStatement getAccIDStmt = null;
+            ResultSet accResultSet = null;
+
+            try{
+                instAccStmt = conn.prepareStatement(
+                        "insert into accounts (username, password) " +
+                                "  values(?, ?) "
+                );
+                instAccStmt.setString(1,username);
+                instAccStmt.setString(2,password);
+
+                instAccStmt.executeUpdate();
+
+                System.out.println(username + "has been added to database");
+
+                getAccIDStmt = conn.prepareStatement(
+                        "select accounts.* from accounts"+
+                                "where username = ? and password = ?"
+                );
+                getAccIDStmt.setString(1,username);
+                getAccIDStmt.setString(2,password);
+
+                List<Account> resultAccounts = new ArrayList<Account>();
 
+                accResultSet = getAccIDStmt.executeQuery();
+
+                //for testing that result was returned i.e. accounts exist
+                Boolean found = false;
+
+                Integer accountID = 0;
+
+                while(accResultSet.next()){
+                    found = true;
+
+                    Account account = new Account();
+                    loadAccount(account, accResultSet, 1);
+                    accountID = account.getAccountID();
+
+                    resultAccounts.add(account);
+                }
+
+                if(!found){
+                    System.out.println("Account was not added successfully");
+                }
+
+                return accountID;
+            }
```

Comment:
Add a catch block for exceptions.
try -> catch - > finally

Reply:
not needed, see previous